### PR TITLE
 Documentation: Fix missing/non-visible doc for Python modules beginning with 'c'

### PR DESCRIPTION
**Pull request by fabratu**
This PR is a follow-up and fixes missing/non-visible doc for modules: clique, coarsening, coloring, components, correlation, csbridge (remaining modules beginning with letter 'c').

### CODE REVIEW

File: networkit/coarsening.pyx [hunk 1]
```diff
@@ -22,19 +22,51 @@ cdef extern from "<networkit/coarsening/GraphCoarsening.hpp>":
...
 
 	def getCoarseGraph(self):
+		"""
+		getCoarseGraph()
+
+		Returns the coarse graph
+
+		Returns
+		-------
+		G
```
Comment (angriman) [inline networkit/coarsening.pyx:41]: Should you specify the return type here? In this case, `networkit.Graph`.

File: networkit/coloring.py [hunk 2]
```diff
@@ -23,6 +55,20 @@ def valid(self, color):
 		return True
 
 	def split(self, color, depth=0):
+		"""
+		split(color, depth=0)
+
+		Helper method for computing spectral coloring. 
+		This function does not need to be called, it is sufficient to call run() method
```
Comment (angriman) [inline networkit/coloring.py:62]: Either `call run()` or `call the run() method`

File: networkit/coloring.py [hunk 3]
```diff
@@ -38,6 +84,13 @@ def split(self, color, depth=0):
 			self.split(otherColor, depth=depth+1)
 
 	def buildReverseDict(self):
+		"""
+		buildReverseDict()
+
+		Helper method for computing spectral coloring. 
+		This function does not need to be called, it is sufficient to call run() method
```
Comment (angriman) [inline networkit/coloring.py:91]: As above

File: networkit/components.pyx [hunk 4]
```diff
@@ -45,78 +45,99 @@ cdef class ConnectedComponents(Algorithm):
...
-		--------
-		map:
+		""" 
+		getComponentSizes()
+		
+		Get the component sizes.
+
+		Returns
+		-------
+		map
```
Comment (angriman) [inline networkit/components.pyx:99]: I think that the return type here should be `dict`, not `map`.

File: networkit/components.pyx [hunk 5]
```diff
@@ -219,22 +278,26 @@ cdef class StronglyConnectedComponents:
...
+		getPartition()
+
 		Returns a Partition object representing the strongly connected components.
 
-		Returns:
-		--------
-		Partition
+		Returns
+		-------
+		networkit.partition
```
Comment (angriman) [inline networkit/components.pyx:287]: This should be `networkit.Partition` (capital `P`)

File: networkit/components.pyx [hunk 6]
```diff
@@ -219,22 +278,26 @@ cdef class StronglyConnectedComponents:
...
+		numberOfComponents()
+		
 		Returns the number of strongly connected components of the graph.
 
-		Returns:
-		--------
-		int
+		Returns
+		-------
+		count
```
Comment (angriman) [inline networkit/components.pyx:300]: I am not sure if we should do this. In the Python docs we often mix C++ types (e.g., `NetworKit::count`) and Python types (e.g., `list`) and this is confusing. However, this issue should be addressed in another PR. For now we should probably avoid using C++ types in Python docs.

File: networkit/components.pyx [hunk 7]
```diff
@@ -243,34 +306,39 @@ cdef class StronglyConnectedComponents:
...
+		Parameters
+		----------
 		u : node
 			A node in the graph.
 
-		Returns:
-		int
+		Returns
+		-----------
+		count
```
Comment (angriman) [inline networkit/components.pyx:316]: As above

File: networkit/components.pyx (hunk 6 above)
Comment (fabratu) [reply networkit/components.pyx:300]: Yes, this makes more sense than mixing both types. Also agree to defer this to another (set of) PR(s). However I will directly use Python types for previously missing doc-strings.

**Comment by fabratu** [comment]
Took care of the comments - thanks for the review!

File: networkit/components.pyx [hunk 8]
```diff
@@ -45,78 +45,99 @@ cdef class ConnectedComponents(Algorithm):
...
+		Get the component sizes.
 
-		Returns:
-		--------
-		map:
-			The map from component to size.
+		Returns
+		-------
+		dict
+			A dict containing the component and their sizes.
```
Comment (angriman) [inline networkit/components.pyx:100]: Maybe it is clearer something like "A dict of 'int: int', mapping component ids to their size."

File: networkit/components.pyx [hunk 9]
```diff
@@ -301,42 +372,59 @@ cdef class WeaklyConnectedComponents(Algorithm):
...
+		Returns the map from component to size.
 
-			Returns:
-			--------
-			map[index, count]
-			 	A map that maps each component to its size.
+		Returns
+		-------
+		dict
+			A dict that maps each component to its size.
```
Comment (angriman) [inline networkit/components.pyx:414]: Each component id.

File: networkit/components.pyx [hunk 10]
```diff
@@ -367,32 +457,41 @@ cdef class BiconnectedComponents(Algorithm):
...
+		Returns the map from component to size.
 
-			Returns:
-			--------
-			map[count, count]
-			A map that maps each component to its size.
+		Returns
+		-------
+		dict
+			A dict that maps each component to its size.
```
Comment (angriman) [inline networkit/components.pyx:481]: Component id to its size.

File: networkit/components.pyx [hunk 11]
```diff
@@ -423,64 +525,87 @@ cdef class DynConnectedComponents(Algorithm):
...
+		Returns the map from component to size.
 
-			Returns:
-			--------
-			map[index, count]
-			 	A map that maps each component to its size.
+		Returns
+		-------
+		dict
+			A dict that maps each component to its size.
```
Comment (angriman) [inline networkit/components.pyx:567]: Component id, as above.

File: networkit/components.pyx [hunk 12]
```diff
@@ -423,64 +525,87 @@ cdef class DynConnectedComponents(Algorithm):
...
+		Returns all the components, each stored as (unordered) set of nodes.
 
-			Returns:
-			--------
-			vector[vector[node]]
-				A vector of vectors. Each inner vector contains all the nodes inside the component.
+		Returns
+		-------
+		list[list[int]]
+			A list of lists. Each inner list contains all the nodes inside the component.
```
Comment (angriman) [inline networkit/components.pyx:580]: In some cases we also use `vector[vector[int]]`. I would recommend using `list` as `vector` is not a Python type.

File: networkit/components.pyx [hunk 13]
```diff
@@ -515,66 +643,88 @@ cdef class DynWeaklyConnectedComponents(Algorithm):
...
+		Returns the map from component to size.
 
-			Returns:
-			--------
-			map[index, count]
-			 	A map that maps each component to its size.
+		Returns
+		-------
+		dict
+			A dict that maps each component to its size.
```
Comment (angriman) [inline networkit/components.pyx:685]: Component id, as above.

File: networkit/components.pyx (hunk 8 above)
Comment (fabratu) [reply networkit/components.pyx:100]: Yeah, we are also not really consistent concerning listing types for lists/dicts. I would maybe prefer something like:
```
dict(int : int)
    A dict, mapping component ids to their size.
```
I am not sure the `,` belongs there btw.

Comment (angriman) [reply networkit/components.pyx:100]: Yes, this is even better, I think. I would not write a comma there.

File: networkit/components.pyx (hunk 12 above)
Comment (fabratu) [reply networkit/components.pyx:580]: Hm, based on the dict-comment above - `list(list(int))` or `list[list[int]]`. The later one would be more in line with Python lists, however in that case we should use `dict{int : int}`, right?

Comment (fabratu) [reply networkit/components.pyx:580]: I think I am leaning more towards using `(` and `)` for naming types of composite data types.

Comment (angriman) [reply networkit/components.pyx:580]: I agree with both.

Comment (fabratu) [reply networkit/components.pyx:580]: So far we had one occasion where `[]` was used (community module). I added this fix to this PR.
